Add support for ROS2 parameters (#7)
orocos/rtt_ros2_integration: rtt_ros2_params/README.md

# Description

> ROS2 provides a slightly different infrastructure to deal with parameters. An introduction guide can be found in [Understanding ROS2 prameters](https://index.ros.org/doc/ros2/Tutorials/Parameters/Understanding-ROS2-Parameters/).

Reviewer: ```suggestion
ROS2 provides a slightly different infrastructure to deal with parameters. An introduction guide can be found in [Understanding ROS2 parameters](https://index.ros.org/doc/ros2/Tutorials/Parameters/Understanding-ROS2-Parameters/).
```

The main difference is that there is no longer a centralized parameter server, but instead, parameters are maintained per node.

> ## `ros2-params` service

Reviewer: For backwards compatibility with [rtt_ros_integration](https://github.com/orocos/rtt_ros_integration) for ROS 1, using service names `rosparam`, `rosservice` and `rosnode` (not in ROS 1) would be another option.

> ## `ros2-params` service

Author: It sounds nice to use the same names `rosparam`, `rosservice`, `rosnode`. But I wonder, in a machine where both ROS 1 and ROS 2 are installed, it could generate some ambiguity. You can only choose which is used based on which package you are importing.

> ## `ros2-params` service

Reviewer: Opened issue #18.

* `storeProperty()`: sets a parameter into the ROS2 node parameter facility from the value of an Orocos property. If the parameter doesn't exists, it declares a new one.

## Usage

This package can be imported with:
```
import("rtt_ros2_params")
```

Then, the service can be loaded into a component named `<component>` with:
```
<component>.loadService('ros2-params')
```

`<component>` will have a new service named `Params` that provides the operations described. These operations can be called with:
```
<component>.Params.setParameter( <args> )
<component>.Params.getParameter( <args> )
<component>.Params.loadProperty( <args> )
<component>.Params.storeProperty( <args> )
```

## Requriements

In order to work, this service requires two other elements being loaded before calling the operations.
* A ROS2 node service must exist for the component where this service is loaded or globally.
* The typekits provided in `rtt_ros2_rclcpp_typekit` and its dependencies must be loaded

These prerequirements can be achieved by:
```
import("rtt_ros2")
import("rtt_ros2_node")
ros.import("rtt_ros2_rclcpp_typekit")

# Optionally

> <component>.loadService("ros2_node")

Author: But, actually, it was not needed to get and set the parameters. I never used this `ros.create_node()` when testing this PR.

> <component>.loadService("ros2_node")

Reviewer: For a component loading the service like that is sufficient and the service constructor will also create the `rclcpp::Node` instance. Only the global node needs to be instantiated explicitly. What I meant is that you need either

```
# Option 1: Instantiate a ROS node for a single component (with name /component_name)
<component>.loadService("ros2_node")
```
or
```
# Option 2: Instantiate a global, process-wide ROS node (with name /rtt_some_random_number)
ros.create_node()
```

But you cannot omit it completely, so it is not optional as the comment suggests. Otherwise [get_ros2_node()](https://github.com/orocos/rtt_ros2_integration/pull/7/files/ab4c541aad7cd022446fc8df0d09e630bfc7b869#diff-bdc8e8b6c7c11afe11fa3688d4ccfa59R268) would fail (to be replaced by the helper methods provided by package `rtt_ros2_node` itself).

What we could do is to also load the `ros2_node` service implicitly into a component whenever the `ros2-params` service is, or to create a global node implicitly on first use of the `ros2-params` service.

`rtt_ros2_params` - ROS parameter support
=========================================

# Description

ROS2 provides a slightly different infrastructure to deal with parameters compared to ROS 1. An introduction guide can be found in [Understanding ROS2 parameters](https://index.ros.org/doc/ros2/Tutorials/Parameters/Understanding-ROS2-Parameters/).
The main difference is that there is no longer a centralized parameter server, but instead, parameters are maintained per node.

## `ros2-params` service

> This package provides a new Orocos service called `ros2_params` that can be loaded globally and per Orocos component. The service provides 4 operations:

Reviewer: ```suggestion
This package provides a new Orocos service called `rosparam` that can be loaded globally and per Orocos component. The service provides 4 operations:
```

* `getParameter()`: loads a parameter from the ROS2 node parameter facility and returns its value.
* `loadProperty()`:  loads a parameter from the ROS2 node parameter facility into an Orocos property. If the property doesn't exist, in creates a new one owned by the `Params` service.
* `setParameter()`: sets a parameter into the ROS2 node parameter facility. If the parameter does not exist, it declares a new one.
* `storeProperty()`: sets a parameter into the ROS2 node parameter facility from the value of an Orocos property. If the parameter doesn't exists, it declares a new one.

## Scripting Interface

This package can be imported with:
```
import("rtt_ros2_params")
```

Then, the service can be loaded into a component named `<component>` with:
```

> <component>.loadService('ros2-params')

Reviewer: ```suggestion
<component>.loadService("rosparam")
```

```

`<component>` will have a new service named `Params` that provides the operations described. These operations can be called with:
```

> <component>.Params.setParameter( <args> )
> <component>.Params.getParameter( <args> )
> <component>.Params.loadProperty( <args> )
> <component>.Params.storeProperty( <args> )

Reviewer: ```suggestion
<component>.rosparam.setParameter( <args> )
<component>.rosparam.getParameter( <args> )
<component>.rosparam.loadProperty( <args> )
<component>.rosparam.storeProperty( <args> )
```

> ## `ros2-params` service

Reviewer: ```suggestion
## `rosparam` service
```

Forgot this one ;-)
